implementations/rust/ockam/ockam_api/src/cli_state/storage/auto_retry.rs

```diff
@@ -0,0 +1,628 @@
+use crate::cli_state::journeys::{Journey, ProjectJourney};
+use crate::cli_state::{
+    EnrollmentsRepository, IdentitiesRepository, IdentityEnrollment, JourneysRepository,
+    NamedIdentity, NamedVault, NodeInfo, NodesRepository, ProjectsRepository, SpacesRepository,
+    TcpInlet, TcpPortalsRepository, UsersRepository, VaultType, VaultsRepository,
+};
+use crate::cloud::email_address::EmailAddress;
+use crate::cloud::enroll::auth0::UserInfo;
+use crate::cloud::project::models::ProjectModel;
+use crate::cloud::space::Space;
+use crate::config::lookup::InternetAddress;
+use crate::nodes::models::portal::OutletStatus;
+use chrono::{DateTime, Utc};
+use ockam::identity::models::{ChangeHistory, CredentialAndPurposeKey, PurposeKeyAttestation};
+use ockam::identity::storage::PurposeKeysRepository;
+use ockam::identity::{
+    ChangeHistoryRepository, CredentialRepository, Identifier, Identity, Purpose,
+    TimestampInSeconds,
+};
+use ockam_core::{async_trait, Address};
+use ockam_vault::storage::SecretsRepository;
+use ockam_vault::{
+    AeadSecret, AeadSecretKeyHandle, SigningSecret, SigningSecretKeyHandle, X25519SecretKey,
+    X25519SecretKeyHandle,
+};
+
+macro_rules! retry {
```

Comment:
Can we instead use a wrapper around `Query::execute` that would retry under the hood, so that we don't need to re-implement all the repository traits?

Reply:
Also, some operations here can consist of multiple queries, and I'm not sure it's guaranteed that all of them are idempotent